### PR TITLE
Fix random splits

### PR DESCRIPTION
**Pull request by pretty-speeches**

### CODE REVIEW

File: src/tabstar/preprocessing/splits.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def do_split(x: DataFrame, y: Series, test_size: int, is_cls: bool, fold: int) -
         return train_test_split(x, y, test_size=test_size, random_state=random_state)
     num_classes = y.nunique()
     if num_classes == 2:
-        return _split_for_binary_classification(x, y, test_size=test_size, random_state=random_state)
+        return _split_for_binary_classification(x, y, test_size=test_size, random_state=random_state) # this function caused the issue
```
Comment (alanarazi7) [inline src/tabstar/preprocessing/splits.py:37]: Yes, makes sense  - it's new. You can remove the comment :)

File: src/tabstar/preprocessing/splits.py [hunk 2]
```diff
@@ -62,9 +62,10 @@ def _split_for_binary_classification(x: DataFrame, y: Series, test_size: int, ra
     test_indices = []
     y_values = sorted(set(y))
     assert len(y_values) == 2, f"Expected 2 classes, got {y_values}"
+    random_number_generator = np.random.RandomState(random_state)
```
Comment (alanarazi7) [inline src/tabstar/preprocessing/splits.py:65]: Interesting. Indeed, this function relies on "random_state" to be passed internally to sklearn's "train_test_split", and we added some randomness. I would have expected this not to be needed, as allegedly fix_seed should determine the whole flow. But, if it works like this, whatever.

**Comment by alanarazi7** [comment]
Very nice work. I have no idea why suddenly they are both aligned, but perhaps that different way of passing the seed is consistent across different environments...?

Re test_tabstar_preprocess, was this not failing before and now does? (I assume it was already broken) 
I would imagine that this feature is affected by randomness as well, since it relies on train/test.

The idea is to see that the preprocessing isn't broken, so given a fold we do train-val, the load the val split and check the values. "assert y[0] == 1" just check that the first row has label 1... But if the train-val changed, this would be wrong. I would recommend just playing around with the values, and finding different rows to change the test. A bit annoying, though, but hopefully this is the last time we change the randomness?

**Review by alanarazi7 — APPROVED** [review]
